packages/eslint-plugin-ckeditor5-rules/lib/rules/no-legacy-imports.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+// Allowed packages to be checked.
+const corePackageName = 'ckeditor5';
+const premiumPackageName = 'ckeditor5-premium-features';
```

Comment:
These two packages are hidden dependencies. The rule does not throw if they do not exist, but also it does not work as expected.

I would mark both of the as [`optionalDependencies`](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#optionaldependencies).

Reply:
Second thought: let's remove it. We do not want to block the release of this package. `npm install treats optional dependencies as dependencies, so it will install it automatically, which isn't our intention. Forgive me for the mess I made.